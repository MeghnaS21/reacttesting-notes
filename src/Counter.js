import { useState } from "react";
function Counter(){
    const [count,setCount]=useState(0)
   
    return(
        <div>
            <h1 data-testid='counter'>{count}</h1>
            <button onClick={increment}>+</button>
            <button onClick={decrement}>-</button>
        </div>
    )
}
export const increment=()=>{
    setCount(count+1)
}
export const decrement=()=>{
    setCount(count-1)
}
export default Counter