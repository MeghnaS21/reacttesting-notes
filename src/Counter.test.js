/*import {render,screen} from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import Counter from "./Counter"
describe('test counter application',()=>{
    test('check increment is working',()=>{
        render(<Counter></Counter>)
        const counter=screen.getByTestId('counter')
        const buttonList=screen.getAllByRole('button')
        userEvent.click(buttonList[0])
        userEvent.click(buttonList[0]) 
        userEvent.click(buttonList[0])
        expect(counter.textContent).toEqual('3')
    })
    test('check decrement is working',()=>{
        render(<Counter></Counter>)
        const counter=screen.getByTestId('counter')
        const buttonList=screen.getAllByRole('button')
        userEvent.click(buttonList[1])
        userEvent.click(buttonList[1]) 
        userEvent.click(buttonList[1])
        expect(counter.textContent).toEqual('-3')
    })
    test('check counter is working',()=>{
        render(<Counter></Counter>)
        const counter=screen.getByTestId('counter')
        const buttonList=screen.getAllByRole('button')
        userEvent.click(buttonList[1])
        userEvent.click(buttonList[0]) 
        userEvent.click(buttonList[1])
        userEvent.click(buttonList[1])
        userEvent.click(buttonList[0])
        expect(counter.textContent).toEqual('-1')
    })
    
})*/
import renderer from 'react-test-renderer'
import Counter,{increment} from './Counter'
    test('matches dom snapshot',()=>{
        const domTree=renderer.create(<Counter></Counter>)
        
        expect(domTree.toJSON()).toMatchSnapshot()
    })
