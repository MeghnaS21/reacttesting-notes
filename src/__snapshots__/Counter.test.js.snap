// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches dom snapshot 1`] = `
<div>
  <h1
    data-testid="counter"
  >
    0
  </h1>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
</div>
`;
